### PR TITLE
Use the latest version of the asciidoctorj plugin

### PR DESCRIPTION
**Pull request by grimreaper**

### CODE REVIEW

**Comment by marcphilipp** [comment]
I've tried this branch locally: Syntax highlighting using rouge does not work in the HTML version of the User Guide. According to @mojavelinux we'll have to wait until asciidoctor 1.7.0 is released.

**Comment by mojavelinux** [comment]
Actually, I corrected that milestone to 1.5.7 (since issues for 1.5.6 are already frozen). Now that I'm familiar with Rouge, it's not going to be difficult to integrate.

File: documentation/documentation.gradle [hunk 1]
```diff
@@ -77,7 +77,7 @@ asciidoctor {
 				'mainDir': project.sourceSets.main.java.srcDirs[0],
 				'testDir': project.sourceSets.test.java.srcDirs[0],
 				'testResourcesDir': project.sourceSets.test.resources.srcDirs[0],
-				'source-highlighter': 'coderay', // TODO switch to 'rouge' once supported by the html5 backend
+				'source-highlighter': 'rouge',
```
Comment (mojavelinux) [inline documentation/documentation.gradle]: The trick here is to set this as:

```
'source-highlighter': 'coderay@'
```

Then in the document header of index.adoc, add an override for PDF:

```
ifdef::backend-pdf[:source-highlighter: rouge]
```

(You could also flip the logic).

**Comment by sbrannen** [comment]
@mojavelinux, that tip looks familiar. I'm sure I've seen it somewhere else _very_ recently. ;-)

I went ahead and implemented that in 2a819ab6d4981396088da725bc06f47d89ef0402.

Cheers!

File: documentation/documentation.gradle (hunk 1 above)
Comment (marcphilipp) [inline documentation/documentation.gradle]: Thanks for chiming in, @mojavelinux! 

What does the `@` do at the end of `coderay@`?

**Comment by sbrannen** [comment]
I am closing this issue since we will be upgrading the Asciidoctor plugin as part of #770 anyway.

File: documentation/documentation.gradle (hunk 1 above)
Comment (sbrannen) [inline documentation/documentation.gradle]: It makes it _overridable_.

Comment (sbrannen) [inline documentation/documentation.gradle]: See Dan's comments in the _related_ [Spring JIRA issue](https://jira.spring.io/browse/SPR-14997?focusedCommentId=136524&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-136524). :wink:

**Comment by sbrannen** [comment]
In the interim, @grimreaper, if you know of any pressing reason to upgrade to 1.5.5 or 1.5.6 (before we upgrade to 1.5.7 in #770) feel free to create new PR that only does that.

**Pull request closed without merging**